assets/js/base/context/cart-checkout/shipping/index.js

```diff
@@ -186,6 +182,7 @@ export const ShippingDataProvider = ( { children } ) => {
 		onShippingRateSelectSuccess,
 		onShippingRateSelectFail,
 		needsShipping,
+		hasError,
```

Comment:
I'm not sure I see the need for splitting out `hasError` from `shippingErrorStatus` (which already contains it).

Reply:
`shippingErrorStatus` is a string with the values `none`, `invalid_address` or `unknown_error`. So it's difficult to use in other contexts because we need to compare that value with the shipping constants.

Alternatively, we could make the shipping context to expose `currentErrorStatus`, which it does expose a `hasError` property. Would that make more sense?

Reply:
oh shoot right, I was confusing `currentErrorStatus` with `shippingErrorStatus`.  I think I had always originally intended to export `currentErrorStatus` as `shippingErrorStatus` and I obviously did not. The intent was not to expose the raw status but just the `is*` equivalents, that way consumers wouldn't need to do it's own conditional checks.

So with that in mind, let's:

- expose currentErrorStatus via aliasing `shippingErrorStatus: currentErrorStatus` in the exported `ShippingData` 
- we should memoize `currentErrorStatus` via `useMemo`.
- switch the emmitters passing along `shippingErrorStatus` to pass along `currentErrorStatus` to observers instead.